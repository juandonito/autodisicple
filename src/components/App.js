import './App.css'

import React from 'react'

import NewTaskBar from './NewTaskBar'
import TaskList from './TaskList'

function App () {
    return (
        <div className='App'>
            <NewTaskBar />
            <TaskList />
        </div>
    )
}

export default App;