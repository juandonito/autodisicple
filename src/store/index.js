import { createStore, applyMiddleware } from 'redux'
import { createLogger } from 'redux-logger'
import rootReducer from '../reducers'

import { loadState, saveState } from '../localStorage'

const logger = createLogger();

const persistedState = loadState();

console.log(persistedState);

const store = createStore(
    rootReducer,
    persistedState,
    applyMiddleware(
        logger
    )
)

store.subscribe(() => {
    saveState({
        ...store.getState()
    });
});

export default store;