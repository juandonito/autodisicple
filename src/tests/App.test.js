import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import store from '../store'
import renderer from 'react-test-renderer'

import App from '../components/App'

describe('App', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <Provider store={store}>
                <App/>
            </Provider>, 
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    })

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Provider store={store}>
                <App/>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

})