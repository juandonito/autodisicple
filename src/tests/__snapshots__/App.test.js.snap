// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="App"
>
  <div
    className="NewTaskBar"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="New task"
        type="text"
        value=""
      />
    </form>
  </div>
  <div
    className="TaskList"
  />
</div>
`;
